src/create_disk.sh

```diff
@@ -8,27 +8,65 @@
 # although see also https://github.com/coreos/coreos-assembler/pull/298
 # For people building "derived"/custom FCOS-like systems, feel free to file
 # an issue and we can discuss configuration needs.
-set -euo pipefail
 
-if [ "$#" -ne 8 ]; then
-	echo 'create_disk <device> <ostree-repo> <ostree-ref> <ostree-remote> <grub-script> <os-name> <save-var-subdirs> <space separated kargs>'
-	exit 1
-fi
+usage() {
+    cat <<EOC
+${0} create a supermin virtual machinge to create a
+Fedora CoreOS style disk image from an OS Tree.
+
+Options:
+    -R|--ostree-ref: the OSTRee reference to install
+    -d|--disk: disk device to use
+    -g|--grub-script: grub script to install
+    -h|--help: show this helper
+    -k|--kargs: kernel CLI args
+    -o|--osname: the OS name to use, e.g. fedora
+    -r|--ostree-remote: the ostree remote
+    -t|--ostree-repo: location of the ostree repo
+    -v|--save-var-subdirs: "yes" to workaround selabel issue for RHCOS
+
+You probably don't want to run this script by hand. This script is
+run as part of 'coreos-assembler build'.
+EOC
+}
+
+while [ $# -gt 0 ];
+do
+    flag="${1}"; shift;
+    arg="${1}"
+    case "${flag}" in
+        -d|--disk)             disk="${arg}"; shift;;
+        -t|--ostree-repo)      ostree="${arg}"; shift;;
+        -R|--ostree-ref)       ref="${arg}"; shift;;
+        -r|--ostree-remote)    remote_name="${arg}"; shift;;
+        -g|--grub-script)      grub_script="${arg}"; shift;;
+        -o|--osname)           os_name="${arg}"; shift;;
+        -v|--save-var-subdirs) save_var_subdirs="${arg}"; shift;;
+        -h|--help)             usage; exit;;
+        -k|--kargs)
+            # any string between -k and -- will be included
+            # as a kernel argument.
+            for arg in ${@}; do
+                if [[ ! "${1}" =~ "--" ]]; then
+                    extrakargs="${extrakargs} ${1}"; shift;
+                fi
+            done;;
+         *) echo "${flag}=${arg} is not understood";;
+         --) break;
+     esac;
+done
 
 export PATH=$PATH:/sbin:/usr/sbin
-
 arch="$(uname -m)"
-
-disk="$1" && shift
-ostree="$1" && shift
-ref="$1" && shift
-remote_name="$1" && shift
-grub_script="$1" && shift
-os_name="$1" && shift
-save_var_subdirs="$1" && shift
-extrakargs="$1" && shift
-
-set -x
+disk="${disk:?--disk must be defined}"
```

Comment:
These wont fail if they're undefined. We should fail hard.

Reply:
Er, howso?  From the bash manual:
`If parameter is null or unset, the expansion of word (or a message to that effect if word is not present) is written to the standard error and the shell, if it is not interactive, exits. Otherwise, the value of parameter is substituted.`

Consider:
```
ben@porpentina.local:/tmp $ cat /tmp/test.sh 
#!/bin/bash
set -x
arg="${arg:?this fails}"
echo did not get here
ben@porpentina.local:/tmp $ bash /tmp/test.sh; echo $?
/tmp/test.sh: line 3: arg: this fails
1
```

Reply:
Oh, got my parameter expansions wrong, LGTM.